contracts/StakingContract.sol

```diff
@@ -262,10 +261,19 @@ contract StakingContract is IStakingContract, IMigratableStakingContract {
 
         emit Withdrew(stakeOwner, res.withdrawnAmount, res.stakedAmount);
 
+        // Trigger state change notifications only in case we're releasing all stakes, thus changing the staking
+        // amounts.
+        if (!releasingAllStakes) {
+            return;
+        }
+
+        assert(res.stakedAmountDiff > 0);
```

Comment:
If Joe has requested unstake all its tokens and later on releasingAllStakes was set. Then when withdrawing the funds, the stakedAmountDiff will be 0. 
We can remove the assert

Reply:
Instead of the assert we can condition the notification if stakedAmountDiff > 0